contracts/Core/BlockManager.sol

```diff
@@ -239,7 +239,7 @@ contract BlockManager is Initializable, ACL, BlockStorage, StateManager {
                 pushAt = i;
                 break;
             }
-            if (proposedBlocks[epoch][i].iteration > iteration) {
+            if (proposedBlocks[epoch][sortedProposedBlockIds[epoch][i]].iteration > iteration) {
```

Comment:
line 238 also needs fix

Reply:
line 238 actually the whole if block is not required

Reply:
addressed here, its not redundant 
https://github.com/razor-network/contracts/issues/414#issuecomment-918263520